src/main/java/me/dinowernli/grpc/polyglot/config/CommandLineArgs.java

```diff
@@ -16,107 +16,95 @@
-  private String methodFilterArg;
-
-  //TODO: Move to a "list_services"-specific flag container
-  @Option(
-      name = "--with_message",
-      metaVar = "true|false",
-      usage="If true, then the message specification for the method is rendered")
-  private String withMessageArg;
-
-  @Option(name = "--metadata", metaVar = "k1:v1,k2:v2,...")
-  private String metadataArg;
+  private final CallCommand callCommand = new CallCommand();
+  private final ListServicesCommand listServicesCommand = new ListServicesCommand();
+
+  @Parameters(separators = "=", commandDescription = "Make a GRPC call to an endpoint")
+  private class CallCommand {
+    @Parameter(names = "--full_method", required = true, description ="<some.package.Service/doSomething>")
+    private String fullMethodArg;
+    @Parameter(names = "--endpoint", required = true, description ="<host>:<port>")
+    private String endpointArg;
+    // The flags below represent overrides for the configuration used at runtime.
+    @Parameter(names = "--use_tls", description ="true|false")
+    private String useTlsArg;
+    @Parameter(names = "--deadline_ms", description ="<number>")
+    private Integer deadlineMs;
+    @Parameter(names = "--tls_ca_cert_path", description ="<path>")
+    private String tlsCaCertPath;
+    @Parameter(names = "--tls_client_cert_path", description ="<path>")
+    private String tlsClientCertPath;
+    @Parameter(names = "--tls_client_key_path", description ="<path>")
+    private String tlsClientKeyPath;
+    @Parameter(names = "--tls_client_override_authority", description ="<host>")
+    private String tlsClientOverrideAuthority;
+    @Parameter(names = "--metadata", description = "k1:v1,k2:v2,...")
+    private String metadataArg;
+  }
 
-  // *************************************************************************
+  @Parameters(separators = "=", commandDescription = "List all known services defined in the proto files")
+  private class ListServicesCommand {
+    @Parameter(names = "--service_filter", description = "Filters service names containing this string e.g. --service_filter TestService")
+    private String serviceFilterArg;
+    @Parameter(names = "--method_filter", description = "Filters service methods to those containing this string e.g. --method_filter List")
+    private String methodFilterArg;
+    @Parameter(names = "--with_message", description = "If true, then the message specification for the method is rendered")
+    private String withMessageArg;
+  }
 
   /**
    * Parses the arguments from the supplied array. Throws {@link IllegalArgumentException} if the
    * supplied array is malformed.
    */
   public static CommandLineArgs parse(String[] args) {
     CommandLineArgs result = new CommandLineArgs();
-    CmdLineParser parser = new CmdLineParser(result);
     try {
-      parser.parseArgument(args);
-    } catch (CmdLineException e) {
+      JCommander jc = JCommander.newBuilder()
```

Comment:
Rather than repeating how to construct a JCommander object multiple times (once here and once for usage), should we perhaps add a CommandLineArgs::asCommander() method?

Reply:
Done